### PR TITLE
Add booted property; make dates nullable and omitempty fields

### PR DESCRIPTION
**Pull request by biancarosa**
# Description

Adds booted property;
Make dates nullable;
Add omiempty fields;

Fixes #THEEDGE-1551

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I run `go fmt ./...` to check that my code is properly formatted
- [x] I run `go vet ./...` to check that my code is free of common Go style mistakes

### CODE REVIEW

File: pkg/services/devices.go [hunk 1]
```diff
@@ -290,3 +291,24 @@ func (s *DeviceService) GetDevices(params *inventory.Params) (*models.DeviceDeta
...
+
+// GetDeviceLastBootedDeployment returns the last booted deployment for a device
+func (s *DeviceService) GetDeviceLastBootedDeployment(device inventory.Device) *inventory.OSTree {
+	var lastDeployment *inventory.OSTree
+	for _, rpmOstree := range device.Ostree.RpmOstreeDeployments {
+		if rpmOstree.Booted {
+			lastDeployment = &rpmOstree
+			break
+		}
+	}
```
Comment (Avielyo10) [inline pkg/services/devices.go:303]:
```suggestion
	for _, rpmOstree := range device.Ostree.RpmOstreeDeployments {
	        rpmOstree := rpmOstree
		if rpmOstree.Booted {
			lastDeployment = &rpmOstree
			break
		}
	}
```

File: pkg/services/devices.go [hunk 2]
```diff
@@ -290,3 +291,24 @@ func (s *DeviceService) GetDevices(params *inventory.Params) (*models.DeviceDeta
...
+	return lastDeployment
+}
+
+// GetDeviceLastDeployment returns the last deployment for a device
+func (s *DeviceService) GetDeviceLastDeployment(device inventory.Device) *inventory.OSTree {
+	var lastDeployment *inventory.OSTree
+	for _, rpmOstree := range device.Ostree.RpmOstreeDeployments {
+		lastDeployment = &rpmOstree
+	}
```
Comment (Avielyo10) [inline pkg/services/devices.go:312]:
```suggestion
	for _, rpmOstree := range device.Ostree.RpmOstreeDeployments {
	        rpmOstree := rpmOstree
		lastDeployment = &rpmOstree
	}
```

**Review by Avielyo10 — COMMENTED** [review]
Try those suggestions to avoid memory aliasing in the for loops

File: pkg/models/base.go [hunk 3]
```diff
@@ -8,8 +8,8 @@ import (
...
 // Model is a basic GoLang struct based on gorm.Model with the JSON tags for openapi3gen
 type Model struct {
-	ID        uint           `gorm:"primarykey" json:"ID,omitempty"`
-	CreatedAt time.Time      `json:"CreatedAt,omitempty"`
-	UpdatedAt time.Time      `json:"UpdatedAt,omitempty"`
-	DeletedAt gorm.DeletedAt `gorm:"index" json:"DeletedAt,omitempty"`
+	ID        uint            `gorm:"primarykey" json:"ID,omitempty"`
+	CreatedAt *time.Time      `json:"CreatedAt,omitempty"`
+	UpdatedAt *time.Time      `json:"UpdatedAt,omitempty"`
+	DeletedAt *gorm.DeletedAt `gorm:"index" json:"DeletedAt,omitempty"`
```
Comment (Avielyo10) [inline pkg/models/base.go:14]:
```suggestion
	DeletedAt gorm.DeletedAt `gorm:"index" json:"DeletedAt,omitempty"`
```
This and `fmt` solves the CI.

Comment (Avielyo10) [reply pkg/models/base.go:14]: My actual suggestion is not using pointers on the DB at all.

**Review by thom-at-redhat — APPROVED** [review]

**Review by maxamillion — APPROVED** [review]

**Pull request merged**